SerialController/PythonCommand.py

```diff
@@ -807,6 +807,114 @@ def do(self):
 			# tm = round(time.time() - start, 2)
 			# print('Loop : {} in {} sec. Average: {} sec/loop'.format(i, tm, round(tm / i, 2)))
 
+class Fossil_shiny(ImageProcPythonCommand):
+	def __init__(self, name, cam):
+		super(Fossil_shiny, self).__init__(name, cam)
+
+	'''
+	head = {0 : カセキのトリ, 1 : カセキのサカナ}
+	body = {0 : カセキのリュウ, 1 : カセキのクビナガ}
+	'''
+	def fossil_loop(self, head=0, body=0):
+		# start = time.time()
+		i = 0
+		while self.checkIfAlive():
+			i += 1
+			for j in range(30):
+				print(str(30*(i-1)+j+1)+"体目, ({}/30 of box)".format(j+1))
+				self.press(Button.A, wait=0.75) #　ん？カセキ そろってんね
+				self.press(Button.A, wait=0.75) # ウカッツの 眼鏡に かなう
+
+				if head == 1:
+					self.press(Direction.DOWN, duration=0.07, wait=0.75) # select fossil
+				self.press(Button.A, wait=0.75) # determine fossil
+
+				if body == 1:
+					self.press(Direction.DOWN, duration=0.07, wait=0.75)  # select fossil
+				self.press(Button.A, wait=0.75) # determine fossil
+
+				self.press(Button.A, wait=0.75) # select "それでよければ"
+				while not self.isContainTemplate('Network_Offline.png', 0.8):
+					self.press(Button.B, wait=0.25)
+					if not self.checkIfAlive(): return
+				self.wait(1.0)
+				if not self.checkIfAlive(): return
+
+			# open up pokemon box
+			self.press(Button.X, wait=1)
+			self.press(Direction.RIGHT, duration=0.07, wait=1)
+			self.press(Button.A, wait=2)
+			self.press(Button.R, wait=2)
+
+			is_contain_shiny = self.CheckBox()
+			# tm = round(time.time() - start, 2)
+			# print('Loop : {} in {} sec. Average: {} sec/loop'.format(i, tm, round(tm / i, 2)))
+			if is_contain_shiny:
+				print('Shiny!')
+				break
+			if not self.checkIfAlive(): return
+
+			self.press(Button.HOME, wait=2)  # EXIT Game
+			self.press(Button.X, wait=0.6)
+			self.press(Button.A, wait=2.5)  # closed
+			self.press(Button.A, wait=1.0)  # Choose game
+			self.press(Button.A)  # User selection
+			while not self.isContainTemplate('OP.png', 0.7): # recognize Opening
+				self.wait(0.2)
+				if not self.checkIfAlive(): return
+			self.press(Button.A, wait=7.0)  # load save-data

```

Comment:
```suggestion
self.press(Button.A)  # load save-data
while not self.isContainTemplate('Network_Offline.png', 0.8):
	self.wait(0.2)
	if not self.checkIfAlive(): return
```

ラグが少し不安なので提供してくださった"Network_Offline.png"を用いた実装はいかがでしょうか